presto-main/src/main/java/com/facebook/presto/metadata/SqlAggregationFunction.java

```diff
@@ -38,6 +38,7 @@
 {
     private final Signature signature;
     private final SqlFunctionVisibility visibility;
+    private final boolean isCalledOnNullInput;
```

Comment:
I think we instead of this field/extra constructor, we can just explicitly override isCalledOnNullInput() only for aggregations where we want to set this to true.

Reply:
That doesn't work for some agg functions like approx_distinct, since they do not derive from SqlAggregationFunction

Reply:
sorry, i meant just for the ones that use SqlAggregationFunction.  The other ones are using the annotations, which looks good.

Reply:
I think I do that already. Look at AbstractMinMaxBy & AbstractMinMaxByNAggregationFunction

Reply:
i guess i'm wondering what is the purpose of this field.  Why is it used for mergeStatisticalDigestFunction, parametricAggregation, and StatistialDigestAggregationFunction.  Why not skip it for mergeStatisticalDigest and StatisticalDigest (since they return false, can use the default). and for parametric aggregation override isCalledOnNullInput() with details.isCalledOnNullInput()? then you can leave out this field entirely. 

Also, as it stands AbstractMinMaxBy overrides the isCalledOnNullInput() method, but this isCalledOnNullInput field within SqlAggregationFunction is set to false. It won't actually be used because you've overridden the method that uses it, but it's still a weird state to be in.

Reply:
Overriding ParametricAggregation makes sense